### PR TITLE
Support for source tmpl files with the ability for string replacement

### PR DESCRIPTION
**Pull request by stevezau**
This PR is to add the ability to have string replacements in .tmpl source files.

This will allow for users to maintain single source files that can be apply across multiple environments.

### CODE REVIEW

**Comment by stevezau** [comment]
This should fix #32

**Comment by abecoffman** [comment]
👍 just gave this a try and works as expected.  You may want to note that you don't have to bind to a host directory to access the `.tmpl` files.  You can copy them into your image at build time.  Other than that, great stuff!

**Comment by stevezau** [comment]
Thanks @abecoffman. I've added some comments re building tmpl files into image.

File: run.sh [hunk 1]
```diff
@@ -13,6 +13,44 @@ if [ -z "$SUMO_ACCESS_ID" ] || [ -z "$SUMO_ACCESS_KEY" ]; then
 	exit 1
 fi
 
+# Support using env as replacement within sources.
+# Gather all template files
+declare -a TEMPLATE_FILES
+if [ -r "${SUMO_SOURCES_JSON}.tmpl" ]; then
+    TEMPLATE_FILES+=("${SUMO_SOURCES_JSON}.tmpl")
+fi
+if [ -r "${SUMO_SYNC_SOURCES}.tmpl" ]; then
```
Comment (maimaisie) [inline run.sh]: I don't think line 22-27 is needed here, `SUMO_SYNC_SOURCES` is a boolean that specifies whether collector should continuously monitor `SUMO_SOURCES_JSON` for automatic configuration update

Comment (bin3377) [inline run.sh]: There may be some misunderstanding? SUMO_SYNC_SOURCES is a boolean value which indicate the source setting is controlled by local file or web config. It's not a file name. So we only need process one file here

File: run.sh [hunk 2]
```diff
@@ -13,6 +13,44 @@ if [ -z "$SUMO_ACCESS_ID" ] || [ -z "$SUMO_ACCESS_KEY" ]; then
...
+    # Replace all env variables and remove .tmpl extension
+    content=`cat ${from}`
+    to=${from%.*}
+
+    while IFS='=' read -r name value ; do
+        content=$(repl_env "${name}" "${value}" "${content}")
+    done < <(env)
+
+    echo "Replacing environment variables from ${from} into ${to}"
+    echo "${content}" > ${to}
```
Comment (bin3377) [inline run.sh]: Open question: Are we sure the current location is writable?

**Comment by stevezau** [comment]
@bin3377 fixed

**Comment by stevezau** [comment]
this should be good now. I didn't update the other run.sh scripts within the folders as i wasnt sure..

File: run.sh (hunk 1 above)
Comment (stevezau) [inline run.sh]: fixed

**Comment by abecoffman** [comment]
Any idea when this might get merged?

**Comment by sumotravisci** [comment]
@abecoffman : could you test with the latest change by Steve?

**Comment by duchatran** [comment]
@abecoffman: Could you test Steve's latest change?

**Comment by bin3377** [comment]
LGTM. Thanks for the fix!

**Comment by maimaisie** [comment]
@stevezau please copy the user.sh file in root dir to docker-sources/run.sh as well

**Comment by stevezau** [comment]
@maimaisie as mentioned ill let you copy the files around as this PR is now closed.

**Pull request merged**